warning.go

```diff
@@ -7,17 +7,26 @@ import (
 	dto "github.com/prometheus/client_model/go"
 )
 
+//go:generate go run golang.org/x/tools/cmd/stringer -linecomment -type=warningCategory -output=warning_string.go
+type warningCategory int
+
+const (
+	warningCategoryUnspecified warningCategory = iota // unspecified
+	getRemoteVersionCategory                          // get_remote_version
```

Comment:
What do you think renaming this to `getRemoteVersionWarningCategory`?
This would make it very explicit what it is used for. Otherwise you always need to look at the type to gain the context in it is used.

Reply:
Or `warningCategoryGetRemoteVersion`? This would be in line with `warningCategoryUnspecified`

Reply:
Renamed to `warningCategoryGetRemoteVersion`.